server/src/test/java/org/elasticsearch/index/seqno/GlobalCheckpointSyncIT.java

```diff
@@ -0,0 +1,213 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.seqno;
+
+import io.crate.common.unit.TimeValue;
+import io.crate.integrationtests.SQLTransportIntegrationTest;
+import org.elasticsearch.action.admin.indices.stats.IndexShardStats;
+import org.elasticsearch.action.admin.indices.stats.IndexStats;
+import org.elasticsearch.action.admin.indices.stats.IndicesStatsResponse;
+import org.elasticsearch.action.admin.indices.stats.ShardStats;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.cluster.node.DiscoveryNodes;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.index.IndexService;
+import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.test.InternalSettingsPlugin;
+import org.elasticsearch.test.transport.MockTransportService;
+import org.elasticsearch.transport.TransportService;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.BrokenBarrierException;
+import java.util.concurrent.CyclicBarrier;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+
+public class GlobalCheckpointSyncIT extends SQLTransportIntegrationTest {
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return Stream.concat(
+            super.nodePlugins().stream(),
+            Stream.of(InternalSettingsPlugin.class, MockTransportService.TestPlugin.class))
+            .collect(Collectors.toList());
+    }
+
+    public void testGlobalCheckpointSyncWithAsyncDurability() throws Exception {
+        internalCluster().ensureAtLeastNumDataNodes(2);
+        execute(
+            "create table test(id integer) clustered into 1 shards with" +
+            "(\"global_checkpoint_sync.interval\" = '1s', \"translog.durability\" = 'ASYNC', \"translog.sync_interval\" = '1s', number_of_replicas = 1)");
+        var indexName = getFqn("test");
+        //ensureGreen(indexName);
+
+        for (int j = 0; j < 10; j++) {
+            execute("insert into " + indexName + "(id) values(?)", new Object[]{j});
+        }
+        //execute("refresh table test");
+        //ensureGreen(indexName);
+
+        assertBusy(() -> {
+            var response = client().admin().indices().prepareStats(indexName).get().getIndex(indexName);
+            assertThat(response, is(notNullValue()));
+            ShardStats[] shards = response.getShards();
+            var seqNoStats = shards[0].getSeqNoStats();
+            assertThat(seqNoStats.getGlobalCheckpoint(), equalTo(seqNoStats.getMaxSeqNo()));
+        });
+    }
```

Comment:
Looks like 

```
var response = client().admin().indices().prepareStats(indexName).get().getIndex(indexName);
```

is broken

If you look at the `get()` result you see the following failures:


```
  shardFailures = DefaultShardOperationFailedException[2]@78  DefaultShardOperationFailedException[]
    0 = DefaultShardOperationFailedException@82 "[euqkzsectldak.test][0] failed, reason [[euqkzsectldak.test/5Em3U1gQQrqFg19JMguc_w][[euqkzsectldak.test][0]] org.elasticsearch.action.support.broadcast.BroadcastShardOperationFailedException: operation indices:monitor/stats failed
        	at org.elasticsearch.action.support.broadcast.node.TransportBroadcastByNodeAction$BroadcastByNodeTransportRequestHandler.onShardOperation(TransportBroadcastByNodeAction.java:420)
        	at org.elasticsearch.action.support.broadcast.node.TransportBroadcastByNodeAction$BroadcastByNodeTransportRequestHandler.messageReceived(TransportBroadcastByNodeAction.java:392)
        	at org.elasticsearch.action.support.broadcast.node.TransportBroadcastByNodeAction$BroadcastByNodeTransportRequestHandler.messageReceived(TransportBroadcastByNodeAction.java:1)
        	at org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:63)
        	at org.elasticsearch.transport.InboundHandler$RequestHandler.doRun(InboundHandler.java:266)
        	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
        	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
        	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
        	at java.lang.Thread.run(Thread.java:832)
        Caused by: java.lang.IllegalStateException: Unknown Flag: Completion
```


I'd suggest to change it to use the `sys.shards` table:


```
+    @Test
     public void testGlobalCheckpointSyncWithAsyncDurability() throws Exception {
         internalCluster().ensureAtLeastNumDataNodes(2);
         execute(
             "create table test(id integer) clustered into 1 shards with" +
             "(\"global_checkpoint_sync.interval\" = '1s', \"translog.durability\" = 'ASYNC', \"translog.sync_interval\" = '1s', number_of_replicas = 1)");
-        var indexName = getFqn("test");
-        //ensureGreen(indexName);

         for (int j = 0; j < 10; j++) {
-            execute("insert into " + indexName + "(id) values(?)", new Object[]{j});
+            execute("insert into test (id) values (?)", new Object[]{j});
         }
-        //execute("refresh table test");
-        //ensureGreen(indexName);

         assertBusy(() -> {
-            var response = client().admin().indices().prepareStats(indexName).get().getIndex(indexName);
-            assertThat(response, is(notNullValue()));
-            ShardStats[] shards = response.getShards();
-            var seqNoStats = shards[0].getSeqNoStats();
-            assertThat(seqNoStats.getGlobalCheckpoint(), equalTo(seqNoStats.getMaxSeqNo()));
+            execute("select seq_no_stats['global_checkpoint'], seq_no_stats['max_seq_no'] from sys.shards where table_name = 'test'");
+            assertThat(response.rowCount(), Matchers.greaterThan(0L));
+            for (var row : response.rows()) {
+                assertThat("global checkpoint and max seq# must be in sync", row[0], is(row[1]));
+            }
         });
     }
```